src/routes/home/__tests__/HomeScreen.spec.tsx

```diff
@@ -29,19 +33,70 @@ jest.mock('../../../components/HeaderWithMenu', () => {
 
 describe('HomeScreen', () => {
   const navigation = createNavigationMock<'Home'>()
+  const nextExercise = {
+    disciplineId: 1,
+    exerciseKey: 2
+  }
+
+  const documents = [
+    {
+      id: 1,
+      word: 'Spachtel',
+      article: ARTICLES[1],
+      document_image: [{ id: 1, image: 'Spachtel' }],
+      audio: 'https://example.com/my-audio',
+      alternatives: [
+        {
+          word: 'Spachtel',
+          article: ARTICLES[2]
+        },
+        {
+          word: 'Alternative',
+          article: ARTICLES[2]
+        }
+      ]
+    },
+    {
+      id: 2,
+      word: 'Auto',
+      article: ARTICLES[1],
+      document_image: [{ id: 1, image: 'Auto' }],
+      audio: '',
+      alternatives: []
+    }
+  ]
 
   it('should render professions', () => {
     mocked(useReadCustomDisciplines).mockReturnValue(getReturnOf([]))
     mocked(useReadSelectedProfessions).mockReturnValue(getReturnOf(mockDisciplines().map(item => item.id)))
     mocked(useLoadDiscipline)
       .mockReturnValueOnce(getReturnOf(mockDisciplines()[0]))
       .mockReturnValueOnce(getReturnOf(mockDisciplines()[1]))
+    mocked(useReadNextExercise).mockReturnValue(getReturnOf(nextExercise))
+    mocked(useReadProgress).mockReturnValue(getReturnOf(0))
     const { getByText } = render(<HomeScreen navigation={navigation} />)
     const firstDiscipline = getByText('First Discipline')
     const secondDiscipline = getByText('Second Discipline')
     expect(firstDiscipline).toBeDefined()
     expect(secondDiscipline).toBeDefined()
-    })
+  })
+
+  it('should navigate to child discipline', () => {
+    mocked(useReadCustomDisciplines).mockReturnValue(getReturnOf([]))
+    mocked(useReadSelectedProfessions).mockReturnValue(getReturnOf(mockDisciplines().map(item => item.id)))
+    mocked(useLoadDiscipline)
+      .mockReturnValue(getReturnOf(mockDisciplines()[0]))
+      .mockReturnValue(getReturnOf(mockDisciplines()[1]))
+    mocked(useReadNextExercise).mockReturnValue(getReturnOf(nextExercise))
+    mocked(useReadProgress).mockReturnValue(getReturnOf(0))
+    React.useState = jest
+      .fn()
+      .mockReturnValue([documents, {}])
+      .mockReturnValue([mockDisciplines()[0], {}])
+      .mockReturnValue([false, {}])
```

Comment:
I am quite sure this is not something one should do :thinking:

Reply:
removed :D